### PR TITLE
Entities, Controllers, Services, DAO: event,event type, image. 

### PR DESCRIPTION
**Pull request by VadimBlagiy**
Pulled base branch in this branch; fixed bug; worked all.

### CODE REVIEW

**Review by ikantsevich — APPROVED** [review]

File: sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/impl/EventServiceImpl.java [hunk 1]
```diff
@@ -1,43 +1,123 @@
...
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Image not found"));
+
+        eventEntity.setImage(imageEntity);
+
+        EventTypeEntity eventTypeEntity = eventTypeDAO.findById(eventUpdateDTO.getEventTypeId())
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Event Type not found"));
+
+        eventEntity.setEventType(eventTypeEntity);
+
+        eventEntity.setEvShortDescription(eventUpdateDTO.getShortDescription());
```
Comment (ikantsevich) [inline sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/impl/EventServiceImpl.java:107]: please use eventMapper here

File: sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/impl/EventServiceImpl.java [hunk 2]
```diff
@@ -1,43 +1,123 @@
...
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Image not found"));
+//
+        eventEntity.setImage(imageEntity);
+
+        EventTypeEntity eventTypeEntity = eventTypeDAO.findById(eventCreateDTO.getEvenTypeId())
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Event Type not found"));
+
+        eventEntity.setEventType(eventTypeEntity);
+
+        eventEntity.setEvShortDescription(eventCreateDTO.getShortDescription());
```
Comment (ikantsevich) [inline sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/impl/EventServiceImpl.java:72]: please use eventMapper here

File: sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/impl/ImageServiceImpl.java [hunk 3]
```diff
@@ -2,42 +2,78 @@
...
+                .collect(Collectors.toList());
     }
 
     @Override
-    public ImageEntity save(ImageEntity imageEntity) {
-        return imageDAO.save(imageEntity);
+    public ImageResponseDTO save(ImageCreateDTO imageCreateDTO) {
+        ImageEntity imageEntity = imageMapper.convertDtoToEntity(imageCreateDTO);
+
+        imageEntity.setImageName(imageCreateDTO.getName());
```
Comment (ikantsevich) [inline sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/impl/ImageServiceImpl.java:51]: the same

File: sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/impl/ImageServiceImpl.java [hunk 4]
```diff
@@ -2,42 +2,78 @@
...
     }
 
     @Override
-    public ImageEntity update(ImageEntity imageEntity) {
-        return imageDAO.save(imageEntity);
+    public ImageResponseDTO update(Long id, ImageUpdateDTO imageUpdateDTO) {
+        ImageEntity imageEntity = imageDAO.findById(id)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Image not found"));
+
+        imageEntity.setImageName(imageUpdateDTO.getName());
```
Comment (ikantsevich) [inline sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/impl/ImageServiceImpl.java:66]: the same

File: sandbox-backend/src/main/java/com/exadel/team2/sandbox/entity/EventTypeEntity.java [hunk 5]
```diff
@@ -20,8 +20,8 @@
     @Column(name = "EVT_ID")
     private Long evtId;
 
-    @OneToOne(mappedBy = "eventType", cascade = {CascadeType.PERSIST, CascadeType.REFRESH} )
-    private EventEntity event;
+//    @OneToOne(mappedBy = "eventType", cascade = {CascadeType.PERSIST, CascadeType.REFRESH} )
+//    private EventEntity event;
```
Comment (ikantsevich) [inline sandbox-backend/src/main/java/com/exadel/team2/sandbox/entity/EventTypeEntity.java:24]: Why it is commented?

File: sandbox-web/src/main/java/com/exadel/team2/sandbox/SandboxApplication.java [hunk 6]
```diff
@@ -1,11 +1,19 @@
...
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
 
 @SpringBootApplication
 public class SandboxApplication {
 
+    @Bean
+    public ModelMapper modelMapper() {
+        return new ModelMapper();
```
Comment (ikantsevich) [inline sandbox-web/src/main/java/com/exadel/team2/sandbox/SandboxApplication.java:14]: Are you using it? Or in all places is used ModelMapper modelMapper = new ModelMapper();

File: sandbox-backend/src/main/java/com/exadel/team2/sandbox/entity/EventTypeEntity.java (hunk 5 above)
Comment (VadimBlagiy) [reply sandbox-backend/src/main/java/com/exadel/team2/sandbox/entity/EventTypeEntity.java:24]: I'll fix

File: sandbox-web/src/main/java/com/exadel/team2/sandbox/SandboxApplication.java (hunk 6 above)
Comment (VadimBlagiy) [reply sandbox-web/src/main/java/com/exadel/team2/sandbox/SandboxApplication.java:14]: Not. I'll remove it.

**Review by akarzhaneuski — APPROVED** [review]

**Pull request merged**